packages/modules/customer/src/models/customer.ts

```diff
@@ -22,8 +27,16 @@ type OptionalCustomerProps =
   | "addresses"
   | DAL.SoftDeletableEntityDateColumns
 
+const CustomerUniqueEmail = createPsqlIndexStatementHelper({
+  tableName: "customer",
```

Comment:
It should be a composite index on email, has_account right?

Reply:
do we accept multiple guest account with the same email?

Reply:
but you are probably right, @olivermrbl ?

Reply:
@adrien2p AFAIK, we accept 1 guest and 1 non-guest account per email.

Reply:
so would one unique on has account true and one on has account false would be more appropriate? wdyt?

Reply:
Like that you can have one of each or if you swith has account to true it works too

Reply:
@adrien2p that would work as well, it just might be a bit more costly performance-wise (on every update, you'll need to update 2 indices, rather than 1) But `UNIQUE INDEX ON (email, has_account)` should have the same effect, right?

Reply:
yes thats right, I ll do that 💪

Reply:
should be good for a last review @sradevski